### PR TITLE
Updating Batch AI

### PR DESCRIPTION
**Pull request by lenala**

### CODE REVIEW

File: src/ResourceManagement/BatchAI/BatchAIJobImpl.cs [hunk 1]
```diff
@@ -290,6 +441,31 @@ public CaffeSettings CaffeSettings()
...
+            return this;
+        }
+
+                ///GENMHASH:AB0BF9D0BEA18CD334AAF69A466D74DB:244EAB58A02E8C0EA8858C0AA028A77B
+        public BatchAIJobImpl WithEnvironmentVariableSecretValue(string name, string keyVaultId, string secretUrl)
+        {
+                    //$ KeyVaultSecretReference secretReference = new KeyVaultSecretReference()
+                    //$ .WithSourceVault(new ResourceId().WithId(keyVaultId)).WithSecretUrl(secretUrl);
+                    //$ ensureEnvironmentVariablesWithSecrets().Add(new EnvironmentVariableWithSecretValue().WithName(name).WithValueSecretReference(secretReference));
+                    //$ return this;
```
Comment (hovsepm) [inline src/ResourceManagement/BatchAI/BatchAIJobImpl.cs]: Why is this code commented? Please either implement or add a text comment with the description why it is not implemented yet.

File: src/ResourceManagement/BatchAI/ContainerImageSettingsImpl.cs [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        private BatchAIJobImpl parent;
+
+        ///GENMHASH:E69EAFBE5FEE03FDD5408C54C9FE8D17:CA53B7BE62B1D1854755EDD010D28892
+        protected  ContainerImageSettingsImpl(ImageSourceRegistry inner, BatchAIJobImpl parent) : base(inner)
+        {
+            this.parent = parent;
+        }
+
+        ///GENMHASH:077EB7776EFFBFAA141C1696E75EF7B3:7EF63673EC66A608A59744FDA0DA741A
+        public IWithCreate Attach()
```
Comment (hovsepm) [inline src/ResourceManagement/BatchAI/ContainerImageSettingsImpl.cs]: Please revise entire file. It does not seems to be implemented properly.

File: src/ResourceManagement/BatchAI/NodeSetupTaskImpl.cs [hunk 3]
```diff
@@ -27,27 +27,55 @@ public NodeSetupTaskImpl WithStdOutErrPath(string stdOutErrPathPrefix)
...
         }
 
+                ///GENMHASH:AB0BF9D0BEA18CD334AAF69A466D74DB:244EAB58A02E8C0EA8858C0AA028A77B
+                public NodeSetupTaskImpl WithEnvironmentVariableSecretValue(string name, string keyVaultId, string secretUrl)
+                {
+                    //$ KeyVaultSecretReference secretReference = new KeyVaultSecretReference()
+                    //$ .WithSourceVault(new ResourceId().WithId(keyVaultId)).WithSecretUrl(secretUrl);
+                    //$ ensureEnvironmentVariablesWithSecrets().Add(new EnvironmentVariableWithSecretValue().WithName(name).WithValueSecretReference(secretReference));
+                    //$ return this;
+
```
Comment (hovsepm) [inline src/ResourceManagement/BatchAI/NodeSetupTaskImpl.cs]: does not seems to be implemented. Also please fix text alignment in your PR.

File: Tests/Fluent.Tests/BatchAI/BatchAITests.cs [hunk 4]
```diff
@@ -129,14 +145,30 @@ public void CanCreateJob()
...
+                OutputDirectory outputDirectory = null;
+//                for (OutputDirectory directory : job.outputDirectories())
+//                {
+//                    if ("OUTPUT".equalsIgnoreCase(directory.id()))
+//                    {
+//                        outputDirectory = directory;
+//                    }
+//                }
+//                Assert.NotNull(outputDirectory);
+//                Assert.Equal("suffix", outputDirectory.PathSuffix.toLowerCase());
```
Comment (hovsepm) [inline Tests/Fluent.Tests/BatchAI/BatchAITests.cs]: code does not seems to be ported properly from Java.

File: src/ResourceManagement/BatchAI/Properties/AssemblyInfo.cs [hunk 5]
```diff
@@ -1,24 +0,0 @@
-// <auto-generated>
```
Comment (hovsepm) [inline src/ResourceManagement/BatchAI/Properties/AssemblyInfo.cs]: Do not remove this file. Please revert.

File: src/ResourceManagement/BatchAI/ContainerImageSettingsImpl.cs (hunk 2 above)
Comment (lenala) [reply src/ResourceManagement/BatchAI/ContainerImageSettingsImpl.cs]: Fixed.

**Comment by lenala** [comment]
Code review addressed.

File: src/ResourceManagement/BatchAI/BatchAIJobImpl.cs [hunk 6]
```diff
@@ -129,8 +233,9 @@ public ToolTypeSettings.CognitiveToolkit.Definition.IBlank<BatchAIJob.Definition
         public override async Task<Microsoft.Azure.Management.BatchAI.Fluent.IBatchAIJob> CreateResourceAsync(CancellationToken cancellationToken = default(CancellationToken))
         {
             createParameters.Cluster = new Models.ResourceId(parent.Id);
-            createParameters.Location = Inner.Location;
-            await Manager.Inner.Jobs.CreateAsync(parent.ResourceGroupName, this.Name, createParameters);
+            createParameters.Location = parent.RegionName;
+            var inner = await Manager.Inner.Jobs.CreateAsync(parent.ResourceGroupName, this.Name, createParameters);
+            SetInner(inner);
```
Comment (hovsepm) [inline src/ResourceManagement/BatchAI/BatchAIJobImpl.cs]: Please add cancellation token parameter to async call.

Comment (hovsepm) [reply src/ResourceManagement/BatchAI/BatchAIJobImpl.cs]: @lenala please address a single comment and create a new PR. I'll merge this to see test results tomorrow.

**Pull request merged**